Update ABL schema to consolidated structure

rnathuji opened the pull request:
This PR ties to the bakery changes in [this PR](https://github.com/openstax/output-producer-service/pull/319).

philschatz commented on README.md:
````diff
@@ -8,37 +8,78 @@
+```json
+[
+  {
+      "repo": "col12067",
````
```suggestion
      "collection_id": "col12067",
```

philschatz commented on README.md:
````diff
@@ -8,37 +8,78 @@
+```json
+[
+  {
+      "repo": "col12067",
````
```suggestion
      "collection_id": "col12067",
```

rnathuji replied:
D'oh! Good 👁️  -- I fixed this.

philschatz reviewed (changes requested):
This PR looks good; I'm kicking myself for suggesting `repo` instead of `repository` or `repository_name` since all of the other fields are full words; what do you & CMs think?

rnathuji commented:
> This PR looks good; I'm kicking myself for suggesting `repo` instead of `repository` or `repository_name` since all of the other fields are full words; what do you & CMs think?

🤔  - Yeah, now that you point it out I see the inconsistency. I'd lean towards changing to one of your suggestions unless CMs definitely want the shorter variant (maybe `repository_name` since it mirrors `collection_id` 🤷‍♂️ ).

rnathuji commented:
@philschatz - This PR has been updated with the following:

* `repo` --> `repository_name` change
* Created the new ABL as `approved-book-list.json` with corresponding `abl-schema.json` so the existing `approved-books.json` and `schema.json` can stay where they were
* Placed `book-slugs.json` back in the repo for the time being just in case CMs find a need to make changes while still using the old `approved-books.json`
* Added a CI test that should fail if changes to `approved-books.json` or `book-slugs.json` are made without a corresponding updated to `approved-book-list.json`
* Added a `convert_abl.py` script to the repo with directions in the `README.md`

philschatz commented on test.js:
```diff
@@ -89,3 +78,123 @@ test('all approved books have a slug defined in book-slugs.json', t => {
+  }, [])
+  t.pass()
+})
+
```
```suggestion

test('approved-book-list.json has at least one version for each book', t => {
  const listData = fs.readFileSync(approvedBookListPath, { encoding: 'utf8' })
  const abl = JSON.parse(listData)
  abl.approved_books.forEach(b => {
    if (b.repository_name) {
      t.truthy(abl.approved_versions.find(v => v.repository_name === b.repository_name), `No version for repository_name=${b.repository_name}`)
    } else {
      t.truthy(abl.approved_versions.find(v => v.collection_id === b.collection_id), `No version for collection_id=${b.collection_id}`)
    }
  })
})

```

It's unlikely to occur but since you added so many thorough tests here's one other corner case

philschatz reviewed (approved):
This looks great! And I think this does not impact REX or CMs workflow
